lib/new_relic/telemetry/broadway.ex

```diff
@@ -0,0 +1,190 @@
+
+  @consumer_start [:broadway, :consumer, :start]
+  @consumer_stop [:broadway, :consumer, :stop]
+
+  def start_link() do
+    enabled = true
+    GenServer.start_link(__MODULE__, [enabled: enabled], name: __MODULE__)
+  end
+
+  def init(enabled: false), do: :ignore
+
+  def init(enabled: true) do
+    config = %{
+      handler_id: :new_relic_broadway
+    }
+
+    :telemetry.attach_many(
+      config.handler_id,
+      [
+        @processor_start,
+        @processor_stop,
+        @message_stop,
+        @message_fail,
+        @consumer_start,
+        @consumer_stop
+      ],
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    Process.flag(:trap_exit, true)
+    {:ok, config}
+  end
+
+  def terminate(_reason, %{handler_id: handler_id}) do
+    :telemetry.detach(handler_id)
+  end
+
+  def handle_event(@processor_start, _measurements, metadata, _config) do
+    Transaction.start_transaction("Broadway", processor_name(metadata))
+    NewRelic.add_attributes(processor_start_attributes(metadata))
+
+    DistributedTrace.generate_new_context()
+    |> DistributedTrace.track_transaction(transport_type: "HTTP")
+  end
+
+  def handle_event(@processor_stop, _measurements, metadata, _config) do
+    NewRelic.add_attributes(processor_stop_attributes(metadata))
+    Transaction.stop_transaction()
+  end
+
+  def handle_event(@message_stop, measurements, metadata, _config) do
+    track_message_segment(measurements, metadata)
+  end
+
+  def handle_event(@message_fail, measurements, metadata, _config) do
+    track_message_segment(measurements, metadata, %{
+      kind: metadata.kind,
+      reason: metadata.reason,
+      stack: metadata.stacktrace
+    })
+  end
+
+  def handle_event(@consumer_start, _measurements, metadata, _config) do
+    Transaction.start_transaction("Broadway", consumer_name(metadata))
+    NewRelic.add_attributes(consumer_start_attributes(metadata))
+  end
+
+  def handle_event(@consumer_stop, _measurements, metadata, _config) do
+    NewRelic.add_attributes(consumer_stop_attributes(metadata))
+    Transaction.stop_transaction()
+  end
+
+  defp extract_callback_module_name(name) do
+    Module.split(name) |> :lists.reverse() |> Enum.drop(2) |> :lists.reverse() |> Enum.join(".")
+  end
+
+  defp processor_name(%{name: name}) do
+    %{"processor_key" => processor_key} =
+      Regex.named_captures(~r/Processor_(?<processor_key>\w+)_\d+$/, to_string(name))
+
+    "#{extract_callback_module_name(name)}/Processor/#{processor_key}"
+  end
+
+  defp processor_start_attributes(%{name: name, messages: messages}) do
+    [
+      "broadway.module": extract_callback_module_name(name),
+      "broadway.stage": :processor,
+      "broadway.processor_message_count": length(messages)
+    ]
+  end
+
+  defp processor_stop_attributes(metadata) do
+    info = Process.info(self(), [:memory, :reductions])
+
+    [
+      "broadway.processor_successful_to_ack_count": length(metadata.successful_messages_to_ack),
```

Comment:
I'm thinking the attribute keys should more closely mirror the names they have in the broadway metadata...

Like `broadway.processor.successful_messages_to_ack` for this one for example

Reply:
Pushed a change here to make a single Transaction around each `handle_message/3`, so this particular piece of code is note relevant anymore.

On `consumer` I had a similar pattern, and applied this changes as requested 👍